pydis_site/apps/api/models/bot/filters.py

```diff
@@ -0,0 +1,187 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models import UniqueConstraint
+
+
+class FilterListType(models.IntegerChoices):
+    """Choice between allow or deny for a list type."""
+
+    ALLOW: 1
+    DENY: 0
+
+
+class InfractionType(models.TextChoices):
+    """Possible type of infractions."""
+
+    NOTE = "Note"
+    WARN = "Warn"
+    MUTE = "Mute"
+    KICK = "Kick"
+    BAN = "Ban"
+
+
+# Valid special values in ping related fields
+VALID_PINGS = ("everyone", "here", "moderators", "onduty", "admins")
+
+
+def validate_ping_field(value_list: List[str]) -> None:
+    """Validate that the values are either a special value or a UID."""
+    for value in value_list:
+        # Check if it is a special value
+        if value in VALID_PINGS:
+            continue
+        # Check if it is a UID
+        if value.isnumeric():
+            continue
+
+        raise ValidationError(f"{value!r} isn't a valid ping type.")
+
+
+class FilterList(models.Model):
+    """Represent a list in its allow or deny form."""
+
+    name = models.CharField(max_length=50, help_text="The unique name of this list.")
+    list_type = models.IntegerField(
+        choices=FilterListType.choices,
+        help_text="Whenever this list is an allowlist or denylist"
+    )
+
+    filters = models.ManyToManyField("Filter", help_text="The content of this list.", default=[])
+    default_settings = models.ForeignKey(
+        "FilterSettings",
+        models.CASCADE,
+        help_text="Default parameters of this list."
+    )
+
+    class Meta:
+        """Constrain name and list_type unique."""
+
+        constraints = (
+            UniqueConstraint(fields=("name", "list_type"), name="unique_name_type"),
+        )
+
+    def __str__(self) -> str:
+        return f"Filter {'allow' if self.list_type == 1 else 'deny'}list {self.name!r}"
+
+
+class FilterSettings(models.Model):
+    """Persistent settings of a filter list."""
+
+    ping_type = ArrayField(
+        models.CharField(max_length=20),
+        validators=(validate_ping_field,),
+        help_text="Who to ping when this filter triggers."
+    )
+    filter_dm = models.BooleanField(help_text="Whenever DMs should be filtered.")
+    dm_ping_type = ArrayField(
+        models.CharField(max_length=20),
+        validators=(validate_ping_field,),
+        help_text="Who to ping when this filter triggers on a DM."
+    )
+    delete_messages = models.BooleanField(
+        help_text="Whenever this filter should delete messages triggering it."
+    )
+    bypass_roles = ArrayField(
+        models.BigIntegerField(),
+        help_text="Roles and users who can bypass this filter."
+    )
+    enabled = models.BooleanField(
+        help_text="Whenever ths filter is currently enabled."
+    )
+    default_action = models.ForeignKey(
+        "FilterAction",
+        models.CASCADE,
+        help_text="The default action to perform."
```

Comment:
Not really clear what that means
```suggestion
        help_text="What action to perform on the triggering user."
```